packages/astro/src/assets/internal.ts

```diff
@@ -89,7 +96,8 @@ export async function generateImage(
 		return undefined;
 	}
 
-	const imageService = (await getConfiguredImageService()) as LocalImageService;
+	// Ensure that the cache directory exists
+	await fs.promises.mkdir(buildOpts.settings.config.cacheDir, { recursive: true });
```

Comment:
We need to make sure we have the permissions to create the directory. If we don't, is the error already handled?

Reply:
They're definitely not, but we don't actually handle any kind of FS error at this stage, even outside assets. 

I'd say that it doesn't really matter and we'd rather the build crash with the error than attempt to continue, so handling it would pretty much just be rethrowing it

Reply:
Good call out. I would expect us to gracefully handle this. If we don't have permission on the cache dir then it should proceed uncached.

Reply:
It does seems superfluous to me considering that: 
- We don't handle not having permissions to `dist` anyway in the end
- In most cases, the user will have just ran `npm install` before running the build, proving that they have write permissions to `node_modules`

But, I can definitely do it if it feels necessary